src/arch/linux/mapping/mapping.c

```diff
@@ -464,6 +464,15 @@ void *mmap_mapping(int cap, dosaddr_t targ, size_t mapsize, int protect)
 
   Q__printf("MAPPING: map, cap=%s, target=%p, size=%zx, protect=%x\n",
 	cap, target, mapsize, protect);
+#ifdef __LP64__
+  if (targ == (dosaddr_t)-1 && _MAP_32BIT == 0) {
```

Comment:
I am not entirely sure if we
need `__LP64__` and `_MAP_32BIT==0`
checks. Maybe it would make sense to
make that code to work universally than
"sometimes"?

Reply:
I mean, even for native DPMI with
MAP_32BIT this still adds an advantage
of avoiding the "negative" pointers that
many progs actually hates.

Reply:
So I propose to do it backwards:
globally enable for now, and later
phase out on 64bit-aware back-ends.
And leave forever on MAP_32BIT
back-ends.

Reply:
Otherwise there is a too big
divergence, i.e. negative pointers
on legacy back-ends, and no such
thing on 64bit ones. Which may lead
to difficult-to-debug problems.